lib/config/configuration.go

```diff
@@ -852,8 +870,8 @@ func applyProxyConfig(fc *FileConfig, cfg *service.Config) error {
 	case legacyKube && newKube:
 		return trace.BadParameter("proxy_service should either set kube_listen_addr/kube_public_addr or kubernetes.enabled, not both; keep kubernetes.enabled if you don't enable kubernetes_service, or keep kube_listen_addr otherwise")
 	case !legacyKube && !newKube:
-		if fc.Version == defaults.TeleportConfigVersionV2 {
-			// Always enable kube service if using config V2 (TLS routing is supported)
+		if fc.Version != defaults.TeleportConfigVersionV1 {
```

Comment:
nit: for all these checks, instead of a `==` or `!=`, I would find a switch easier to reason about.

```
switch fs.Version {
case defaults.TeleportConfigVersionV1:
    // v1 logic
case defaults.TeleportConfigVersionV2, defaults.TeleportConfigVersionV3:
    // v2/v3 logic
}
```

Reply:
I had chose to do it this way so we have less places to update every time we do a new config version, happy to change it to a switch statement though